src/internal/connector/onedrive/collections.go

```diff
@@ -366,6 +367,56 @@ func (c *Collections) Get(
 	return collections, excludedItems, nil
 }
 
+func updateCollectionPaths(
+	id string,
+	cmap map[string]data.BackupCollection,
+	newPaths map[string]string,
+	curPath path.Path,
+) (bool, error) {
+	var initialCurPath path.Path
+
+	col, found := cmap[id]
+	if found {
+		ocol, ok := col.(*Collection)
+		if !ok {
+			return found, clues.New("unable to cast onedrive collection")
+		}
+
+		initialCurPath = ocol.FullPath()
+		if initialCurPath.String() == curPath.String() {
+			return found, nil
+		}
+
+		ocol.SetPaths(ocol.PreviousPath(), curPath)
+	}
+
+	if initialCurPath == nil {
+		return found, nil
+	}
+
+	for i, c := range cmap {
+		if i == id {
+			continue
+		}
+
+		ocol, ok := c.(*Collection)
+		if !ok {
+			return found, clues.New("unable to cast onedrive collection")
+		}
+
+		colPath := c.FullPath()
+
+		// Only updates if initialCurPath parent of colPath
+		updated := colPath.UpdateParent(initialCurPath, curPath)
+		if updated {
+			ocol.SetPaths(c.PreviousPath(), colPath)
+			updatePath(newPaths, i, colPath.String())
```

Comment:
this doesn't need to be called. `updatePath` also updates subfolder paths for the given path and it's already called when we handle the move for the collection
```suggestion
```